### PR TITLE
[SPARK-23886][SS] Update query status for ContinuousExecution

### PR DESCRIPTION
**Pull request by spaced4ndy**
## What changes were proposed in this pull request?

Added query status updates to ContinuousExecution

## How was this patch tested?

Existing unit tests, added a test to StreamingQueryStatusAndProgressSuite

### CODE REVIEW

**Comment by efimpoberezkin** [comment]
@jose-torres Hi Jose, I've added a couple of update messages to places you suggested, query stop indeed seems to be reported in StreamExecution.
I have a question regarding updating isDataAvailable field of query status. In MicroBatchExecution it is updated based on the result of comparison of committed and available offsets, and the latter is being changed in constructNextBatch(). Would the same approach be valid for ContinuousExecuction and if yes how should available offsets be updated?

**Comment by jose-torres** [comment]
I'm not sure isDataAvailable makes sense in the context of continuous processing; it seems fundamentally tied to the microbatch execution model. I think the best option is to just leave it and isTriggerActive always false, and take a TODO to restructure StreamingQueryStatus to eliminate the assumption they're meaningful. (The latter would be an API change, so we'd definitely want a separate PR for it - fortunately StreamingQueryStatus isn't stable yet.)

**Comment by jose-torres** [comment]
I guess we might not even need to make an API change, just document that these flags only mean anything for microbatch execution. In any case that's a separate discussion.

**Comment by efimpoberezkin** [comment]
Hi @jose-torres, I made some changes to this PR according to your comment, could you review it please?

**Comment by jose-torres** [comment]
The approach looks good to me, but we probably want to add some tests to StreamingQueryStatusAndProgressSuite. (See test("basic") in ContinuousSuite for how to set up a continuous processing memory stream, and note that processAllAvailable() won't work properly for continuous execution - you'll want to use CheckAnswer to await the added data and Execute to do the test-specific progress checks)

File: sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/ProgressReporter.scala [hunk 1]
```diff
@@ -111,7 +112,12 @@ trait ProgressReporter extends Logging {
     logDebug("Starting Trigger Calculation")
     lastTriggerStartTimestamp = currentTriggerStartTimestamp
     currentTriggerStartTimestamp = triggerClock.getTimeMillis()
-    currentStatus = currentStatus.copy(isTriggerActive = true)
+    // isTriggerActive field is kept false for ContinuousExecution
+    // since it is tied to MicroBatchExecution
+    this match {
```
[the remark this answers is not in the document]
Comment (efimpoberezkin) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/ProgressReporter.scala]: Yep, an oversight on my part. Fixed it.

[the remark this answers is not in the document]
Comment (attilapiros) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/ProgressReporter.scala]: For me a bit strange that `ProgressReporter` as a base for `MicroBatchExecution` and `ContinuousExecution` contains an implementation which is only good for `MicroBatchExecution` and overriden only in `ContinuousExecution`. I mean why providing an implementation here which is only good for one of the inherited class and why not letting it to be just an abstract method here?

**Pull request closed without merging**